src/inmanta/deploy/scheduler.py

```diff
@@ -644,6 +782,11 @@ async def _new_version(
         requires: Mapping[ResourceIdStr, Set[ResourceIdStr]],
         up_to_date_resources: Optional[Set[ResourceIdStr]] = None,
         reason: str = "Deploy was triggered because a new version has been released",
+        start_deployment: bool = True,
+        intent_lock_acquired: bool = False,
+        scheduler_lock_acquired: bool = False,
```

Comment:
I find that all these parameters make it a lot more difficult to reason on what's going on and how this may interact with other scheduler flows. But I do think much of these parameters can be dropped when we (I?) rework the non-initialize new-version flow to also use the iterative process, so I guess we can leave it be for now.